### ISSUE TITLE
PC not staying connected to network when offline 

### ISSUE THREAD

**Issue by Benjamin8obadiah** [ClassicOldSong/Apollo#184]
i know the program technically runs on lan but whenever my router goes offline it no longer works (host and target still connected to network). there are some online requirements. can these be removed?

**Comment by ClassicOldSong** [comment]
There isn't any online requirements. Can you describe what the symptoms are when it's not working and what's your client?

If you added your host manually using its public IP then when your router goes offline it can't see that IP anymore. In that case it won't connect, you'll just need to add your host using its LAN address again.

**Comment by Benjamin8obadiah** [comment]
There is one wifi 6 router, plugged into Internet source. The PC is wired with Ethernet to the router, and the target (s24 ultra smartphone) is connected wirelessly. 

This setup works, while the router is supplied internet. As soon as the routers internet supply goes out, the stream cuts off with this error code;

_The connection was terminated. Error code: -1_

The phone remains connected to the router, as it's set to "remain connected without Internet" 

As I'm typing this I think I figured out what the problem is. The PC is disassociating itself from the Ethernet that's plugged into it. It requires an internet connection from the plug, and when it doesn't have one, it just disconnects. And doesn't remain connected on lan. 

I'm sure there's a simple setting I just don't know about to fix this. One similar to the android settings to remain connected while offline.

**Comment by ClassicOldSong** [comment]
Are you using IPv6 or IPv4? If you're using the v6 address assigned from your ISP provider, then disconnecting from WAN could cause problems. If you setup IPv4 and use NAT on your router, there should be no problems.

**Comment by ClassicOldSong** [comment]
https://en.wikipedia.org/wiki/Network_address_translation

Usually you don't need to manually set NAT on your router if you're using IPv4. Can you long press on the computer icon on your client and check its detail, make sure it is using a local IP address?

Also how does the WAN go out? Pulling the cable or disconnecting via the router's admin console?

**Comment by kayti** [comment]
Check if Your phone is REALLY trying to use Your WiFi (example ping router form it). I had many cases where phone was "connected" to wifi without internet but couldn't access LAN anyway because it was directly switching to Mobile network.

**Comment by Benjamin8obadiah** [comment]
> https://en.wikipedia.org/wiki/Network_address_translation
> 
> Usually you don't need to manually set NAT on your router if you're using IPv4. Can you long press on the computer icon on your client and check its detail, make sure it is using a local IP address?
> 
computer is using an ip  like 192.168.x.x.

> Also how does the WAN go out? Pulling the cable or disconnecting via the router's admin console?

pulling the cable

**Comment by Benjamin8obadiah** [comment]
> Check if Your phone is REALLY trying to use Your WiFi (example ping router form it). I had many cases where phone was "connected" to wifi without internet but couldn't access LAN anyway because it was directly switching to Mobile network.

i confirm that my phone is still connected to the router. i log into the routers web interface, and can control it.  mobile data is off.

i can do that from my phone (target device) but cant do it from the host pc. 

the usual ethernet logo at the bottom right of windows 11 is replaced with the no internet logo.

**Comment by Benjamin8obadiah** [comment]
the ethernet controller, by "Allow the computer to turn off this device to save power

**Comment by Benjamin8obadiah** [comment]
@Wordsmith9091 yes, you are correct. This is an issue with my PC not staying connected to my network when the network has no Internet connection. Any tips on how to solve this?

**Comment by Wordsmith9091** [comment]
@Benjamin8obadiah Sorry, I haven't run into that before. If your PC has any vendor software installed that's meant to manage the network connection, I'd check for options in that (and possibly just uninstall it).

**Comment by Benjamin8obadiah** [comment]
> @Benjamin8obadiah Sorry, I haven't run into that before. If your PC has any vendor software installed that's meant to manage the network connection, I'd check for options in that (and possibly just uninstall it).

there is no vendor software causing it. and i can confirm that this is an ethernet only issue.

when the pc is connected via wifi to this network and the network has no internet, it stays connected. 

and i can access the router webpage from my pc.
